core/controllers/practice_sessions_test.py

```diff
@@ -132,30 +135,28 @@ def test_get_fails_when_skill_ids_dont_exist(self):
     def test_any_user_can_access_practice_sessions_data(self):
         with self.swap(constants, 'ENABLE_NEW_STRUCTURE_PLAYERS', True):
             json_response = self.get_json(
-                '%s/%s' % (
+                '%s/%s?selected_subtopics=["subtopic_name"]' % (
                     feconf.PRACTICE_SESSION_DATA_URL_PREFIX,
                     'public_topic_name'))
             self.assertEqual(json_response['topic_name'], 'public_topic_name')
-            self.assertEqual(len(json_response['skill_descriptions']), 2)
             self.assertEqual(
-                json_response['skill_descriptions']['skill_id_1'],
-                'Skill 1')
+                len(json_response['skill_ids_to_descriptions_map']), 1)
             self.assertEqual(
-                json_response['skill_descriptions']['skill_id_2'],
+                json_response['skill_ids_to_descriptions_map']['skill_id_2'],
                 'Skill 2')
 
     def test_no_user_can_access_unpublished_topic_practice_session_data(self):
         with self.swap(constants, 'ENABLE_NEW_STRUCTURE_PLAYERS', True):
             self.get_json(
-                '%s/%s' % (
+                '%s/%s?selected_subtopics=["subtopic_name"]' % (
                     feconf.PRACTICE_SESSION_DATA_URL_PREFIX,
                     'private_topic_name'),
                 expected_status_int=404)
 
     def test_get_fails_when_topic_doesnt_exist(self):
         with self.swap(constants, 'ENABLE_NEW_STRUCTURE_PLAYERS', True):
             self.get_json(
-                '%s/%s' % (
+                '%s/%s?selected_subtopics=["subtopic_name"]' % (
```

Comment:
Just a question about this. Do we have the concept of subtopic ids? If so, any reason not to use those instead? It might look better in the url than a list (and if the ids are from a restricted charset then the issues with comma-separation might go away).

Reply:
Yeah, changed it to use comma separated subtopic IDs